pkg/kubernetes/provider.go

```diff
@@ -155,3 +140,32 @@ func sortProviders(providers map[string]struct{}) []string {
 
 	return sortedProviders
 }
+
+// Reset overwrites all providers in the provider store given a list of providers
+func (p *ProviderStore) Reset(providersList map[string]struct{}) map[string]struct{} {
+	p.mutex.Lock()
```

Comment:
nit - since we dropped Watch mechanism and are generating query list on every reconcile, is in memory `ProviderStore` struct and it's synchronized state handling still needed?